### PR TITLE
set umask for pkgdb creation/access 

### PR DESCRIPTION
**Pull request by datenwolf**
This commit adds setting the process umask to 0664 prior to opening/creating so that pkgdb remains readable for world, even if xbps tools have been called with a very restrictive umask (like through sudo by a user with a umask 0700).

Maybe this behavior should be settable through a configuration option or a environment variable.

### CODE REVIEW

File: lib/pkgdb.c [hunk 1]
```diff
@@ -89,13 +91,18 @@ xbps_pkgdb_lock(struct xbps_handle *xhp)
 			return rv;
 		}
 	}
+
+	prev_umask = umask(0644);
```
Comment (xtraeme) [inline lib/pkgdb.c]: This should be `umask(022)` instead.

Comment (datenwolf) [inline lib/pkgdb.c]: Of course. Little brainfart on my side.

**Comment by datenwolf** [comment]
FYI I fixed that inverted umask value. Would be great if this could be merged upstream, since so far I'm using my own xbps build, for that very pkgdb permissions reason.

**Comment by xtraeme** [comment]
Please rebase and submit a proper commit log message.

**Comment by avoidr** [comment]
This problem could have been effectively solved by putting "Defaults umask_override" in /etc/sudoers.

It isn't a xbps problem. Try mounting something with sudo, you'll get the same result. Or, say, touching something should get you more restrictive permissions than if you did it as root (assuming umask 0022) directly.

**Comment by anddam** [comment]
Is there a reason why the sudo umask override option cannot be set in default sudoers on a fresh installed system?

**Comment by Duncaen** [comment]
Because its not the correct solution, there are other tools than sudo, and you are free to use whatever umask you like, and xbps should handle it.

**Pull request merged**